### PR TITLE
using authenticate to access Github package name. fixes #1262

### LINKED ISSUE #1262: Error resolving package name for private github repo
I'm running into some problems when installing and resolving dependencies for packages that imports/depends on packages in a private github repo.

``` r
> deps <- devtools::dev_package_deps(dependencies = TRUE)
Using GitHub PAT from envvar GITHUB_PAT
Using GitHub PAT from envvar GITHUB_PAT
Error in data.frame(package = package, installed = installed, available = available,  : 
  row names contain missing values
> 
```

The issue seems to be that we are not authenticating to Github when trying to resolve the package name. We then end up getting an error message when trying to construct a `data.frame` of named vectors with `NA` names.

I'm submitting a pull request for a solution where we:
1. Use the oath_token to authenticate before downloading the DESCRIPTION file.
2. Download the description from https://raw.githubusercontent.com/user/repo/... instead of https://github.com/user/repo/RAW/.... For some reason the latter approach doesn't seem to work.

### PR DESCRIPTION
**Pull request by eriknil**
The new version of devtools is unable to resolve remote package names for private github repositories.

Used same approach as in `remote_download.github_remote` for
authenticating, and changed the url used from
https://github.com/user/repo/RAW/... to
https://raw.githubusercontent.com/user/repo/...

This should fix #1262

### CODE REVIEW

File: R/install-github.r [hunk 1]
```diff
@@ -302,3 +311,4 @@ download_github <- function(path, url, ...) {
   writeBin(httr::content(request, "raw"), path)
   path
 }
+
```
Comment (jimhester) [inline R/install-github.r]: Could you get rid of the trailing line here

**Comment by eriknil** [comment]
of course

**Comment by hadley** [comment]
@jimhester is this good to merge?

**Comment by jalazawa** [comment]
I have the same issue

**Comment by jimhester** [comment]
Thanks! Merged in a61d87fc995

**Pull request closed without merging**